src/casing.f90

```diff
@@ -249,265 +213,382 @@ subroutine casing( teta, zeta, gBn, icasing )
 
 end subroutine casing
 
+!> \brief Compute the field produced by the plasma currents, at an arbitrary, external location using virtual casing.
+!> \ingroup grp_free-boundary
+!>
+!> casing2() computes the virtual casing field using a fixed resolution grid on the plasma boundary, in contrast to casing() which uses an adaptive integration routine.
+!> Because the evaluation positions are known in advance, the most expensive terms of the computation (dvcfield) are precomputed. 
+!> The integral computed by casing2() is 
+!> \f{eqnarray}{ \int_0^{2\pi} \int_0^{2\pi} \frac{{\bf j}(\theta, \zeta) \times {\bf D}(\theta, \zeta)}{\|{\bf D}(\theta', \zeta')\|^3}+ \;\; d\theta d\zeta  
+!>  \f}
+!> with \f${\bf r}\f$ approximately the distance vector from an evaluation point of the current surface \f${\bf x}(\theta, \zeta)\f$ to the 
+!> external point \f$(x,y,z)\f$, and the exact formula includes a regularization factor \f$\epsilon\f$  Eqn.\f$(\ref{eq:vcasing_distance})\f$..
+subroutine casing2( xyz, nxyz,  Pbxyz, Jxyz, vcstep,gBn)
+  use constants, only : zero, one, three, pi2
+
+  use fileunits, only : ounit, vunit
+
+  use inputlist, only : vcasingeps, vcNz, vcNt
+
+  use allglobal, only : myid, ncpu, cpus, MPI_COMM_SPEC !, Pbxyz, Jxyz
+
+  LOCALS
+
+  REAL, intent(in)     :: xyz(3) ! arbitrary location; Cartesian;
+  REAL, intent(in)     :: nxyz(3) ! surface normal on the computational boundary; Cartesian;
+  REAL, intent(in)     :: Pbxyz(1:vcNz*vcNt, 1:3), Jxyz(1:vcNz*vcNt, 1:3) 
+  INTEGER, intent(in)  :: vcstep
+  REAL, intent(out)    :: gBn ! B.n on the computational boundary;
+  
+  REAL :: rr(1:3),  distance(1:3), jj(1:3), Bxyz(1:3), accumulator, firstorderfactor
+  INTEGER :: plasmaNtz, jk
+
+!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!
+  ! plasmaNtz = SIZE(Pbxyz, 1)
+  plasmaNtz = vcNz*vcNt
+  ! loop over the high resolution plasma boundary (inner boundary for virtual casing)
+  do jk = 1, plasmaNtz, vcstep ;
+      ! position on computational boundary - position on plasma boundary
```

Comment:
nit: I understand the `-` here as difference; what about writing `distance between point on computational boundary and point on plasma boundary` to make this more clear?